climada_petals/engine/warn.py

```diff
@@ -0,0 +1,191 @@
+import logging
+
+import numpy as np
+import skimage
+import copy
+from dataclasses import dataclass
+
+LOGGER = logging.getLogger(__name__)
+
+
+@dataclass
+class FilterData:
```

Comment:
Is this class really needed? It looks to me that you never pass an object of this class to a method... Why would you need it then? 

@emanuel-schmid : what do you think, is this dataclass useful?

Reply:
The way I understand it, the `FilterData` is just a compilation of Parameters that are used throughout the `Warn.from_map` "constructor".
As such, I think the class has its justification. Storing a set of parameters for later use can be convenient.

However
- the class name may be more to the point, like `WarnParameters` or something
- as it most likely won't be used outside of this particular context one could think about moving it inside the `Warn` class
- though by nature it is most certainly one, it's not implemented as a dataclass. Therefore I suggest to remove that tag or, preferably, implement it as a dataclass (which, among other things, involves importing `List` from `typing`, see https://stackoverflow.com/questions/57690749/how-to-declare-an-array-or-a-list-in-a-python-dataclass)

Reply:
Thanks for that feedback. I made these changes and I also set the defaults now differently (as it is done in dataclasses, right?). Is this better now?

Reply:
@emanuel-schmid I would be very happy if you could have a quick look at the newly designed data class. Is this now better and can I improve it further?